### PR TITLE
Replace magic numbers with named state constants

### PR DESCRIPTION
**Pull request by swlynch99**
I think I've gotten the numbers right, but please take a second look just to be sure everything is correct.

### CODE REVIEW

File: src/k_lock_mutex.rs [hunk 1]
```diff
@@ -210,26 +214,26 @@ impl<T: ?Sized> Mutex<T> {
...
 
-            if (state == 0
+            if (state == UNLOCKED
                 && self
                     .futex
-                    .compare_exchange(0, 1, Ordering::Acquire, Ordering::Relaxed)
+                    .compare_exchange(UNLOCKED, LOCKED, Ordering::Acquire, Ordering::Relaxed)
                     .is_ok())
-                || (state != 2 && self.futex.swap(2, Ordering::Acquire) == 0)
+                || (state != CONTENDED && self.futex.swap(CONTENDED, Ordering::Acquire) == 0)
```
Comment (kvc0) [inline src/k_lock_mutex.rs]:
```suggestion
                || (state != CONTENDED && self.futex.swap(CONTENDED, Ordering::Acquire) == UNLOCKED)
```
The 0 in this case has significant meaning

File: src/k_lock_mutex.rs [hunk 2]
```diff
@@ -372,7 +376,7 @@ impl<T: ?Sized> DerefMut for MutexGuard<'_, T> {
 impl<T: ?Sized> Drop for MutexGuard<'_, T> {
     #[inline]
     fn drop(&mut self) {
-        if self.lock.futex.swap(0, Ordering::Release) == 2 {
+        if self.lock.futex.swap(UNLOCKED, Ordering::Release) == 2 {
```
Comment (kvc0) [inline src/k_lock_mutex.rs]:
```suggestion
        if self.lock.futex.swap(UNLOCKED, Ordering::Release) == CONTENDED {
```

**Review by kvc0 — COMMENTED** [review]
just a couple nits for completion

**Comment by swlynch99** [comment]
Alright, that should be all of them now

**Review by kvc0 — APPROVED** [review]
This looks great, thanks!

**Pull request merged**